src/backend/marsha/core/views.py

```diff
@@ -547,11 +594,15 @@ def _get_app_data(self, lti):
 
         lti_select_form_data = self.request.POST.copy()
         lti_select_form_data["lti_message_type"] = "ContentItemSelection"
+
+        jwt_token = AccessToken()
+        jwt_token.payload["lti_select_form_data"] = lti_select_form_data
+
         app_data.update(
             {
                 "frontend": "LTI",
                 "lti_select_form_action_url": reverse("respond_lti_view"),
-                "lti_select_form_data": lti_select_form_data,
+                "lti_select_form_data": {"jwt": str(jwt_token)},
```

Comment:
Why including again the token in the response ? The response will not be used anymore by marsha if I understand and you will in the js part decode the token to retrieve the info you need in order to send them to moodle.

Reply:
In fact this is the first view called, the token is the only data sent to the frontend.
After clicking on the chosen content, the token is send to marsha with the selected content.
Then on the respond view, the token is decoded, his payload along with selected content are signed, and used to build a form which is sent back to moodle.
The token is never decoded on front, nor altered.

Reply:
Sorry, I should specify all this in the PR description, I'll do it now.